Tools/Scripts/libraries/webkitcorepy/webkitcorepy/file_lock.py

```diff
@@ -22,60 +22,86 @@
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 import os
+import re
 import sys
 import time
 
+from webkitcorepy import string_utils
+
 if sys.platform.startswith('win'):
     import msvcrt
-else:
-    import fcntl
 
 
 class FileLock(object):
+    INTEGER_RE = re.compile(r'^\d+$')
+    USE_WINDOWS = sys.platform.startswith('win')
+    USE_EXLOCK = not USE_WINDOWS and getattr(os, 'O_EXLOCK', False)
+
+    @classmethod
+    def is_process_running(cls, pid):
+        try:
+            os.kill(pid, 0)
+            return True
+        except OSError:
+            return False
```

Comment:
Maybe check if the error is `errno.ESRCH`. I'm a little concerned about some of the autoinstallers running under `sudo`. But not an issue when building.

Reply:
We already handle auto installers running under `sudo`, actually, we make sure that the entire directory is owned by the user after we finish installing a package. That will include this file. It's not a perfect system (if we manage to have Python killed when doing an install, pretty sure we'd leave some `sudo` files behind), but it's going to be pretty effective.